OptimizelySDKCore/OptimizelySDKCore/OPTLYVariation.m

```diff
@@ -16,14 +16,47 @@
 
 #import "OPTLYVariation.h"
 #import "OPTLYDatafileKeys.h"
+#import "OPTLYVariableUsage.h"
+
+@interface OPTLYVariation()
+/// A mapping of Feature Variable IDs to Variable Usages constructed during the initialization of Variation objects from the list of Variable Usages.
+@property (nonatomic, strong) NSDictionary<NSString *, OPTLYVariableUsage *><Ignore> *variableIdToVariableUsageMap;
+@end
 
 @implementation OPTLYVariation
 
 + (OPTLYJSONKeyMapper*)keyMapper
 {
-    return [[OPTLYJSONKeyMapper alloc] initWithDictionary:@{ OPTLYDatafileKeysVariationId    : @"variationId",
-                                                        OPTLYDatafileKeysVariationKey   : @"variationKey"
+    return [[OPTLYJSONKeyMapper alloc] initWithDictionary:@{ OPTLYDatafileKeysVariationId   : @"variationId",
+                                                             OPTLYDatafileKeysVariationKey  : @"variationKey",
+                                                             OPTLYDatafileKeysVariationVariableUsageInstances  : @"variableUsageInstances"
                                                        }];
 }
 
+# pragma mark - Feature Variable Mappings and Getters
+
+- (nullable OPTLYVariableUsage *)getVariableUsageForVariableId:(nullable NSString *)variableId {
+    OPTLYVariableUsage *variableUsage = nil;
+    if (variableId) {
+        variableUsage = self.variableIdToVariableUsageMap[variableId];
+    }
+    return variableUsage;
+}
+
+- (NSDictionary<NSString *, OPTLYVariableUsage *> *)variableIdToVariableUsageMap {
+    if (!_variableIdToVariableUsageMap) {
+        _variableIdToVariableUsageMap = [self generateVariableIdToVariableUsageMap];
+    }
+    return _variableIdToVariableUsageMap;
+}
+
+- (NSDictionary<NSString *, OPTLYVariableUsage *> *)generateVariableIdToVariableUsageMap {
+    NSMutableDictionary *map = [[NSMutableDictionary alloc] init];
+    for (OPTLYVariableUsage *variableUsage in self.variableUsageInstances) {
+        map[variableUsage.variableId] = variableUsage;
+    }
+    
+    return [NSDictionary dictionaryWithDictionary:map];
+}
```

Comment:
MISSING UNIT TESTS FOR THESE NEW METHODS?

Are we missing Xcode unit tests for the three new methods
```
- (nullable OPTLYVariableUsage *)getVariableUsageForVariableId:(nullable NSString *)variableId {...}
- (NSDictionary<NSString *, OPTLYVariableUsage *> *)variableIdToVariableUsageMap {...}
- (NSDictionary<NSString *, OPTLYVariableUsage *> *)generateVariableIdToVariableUsageMap {...}
```
?

I ran OptimizelySDKCoreiOSTests with "Code Coverage" turned on.  These 3 new methods
weren't called in the "Show the Report navigator" "Coverage" report for OPTLYVariation.m
which is now only at 22.22% coverage.

Reply:
Yes. These 3 new methods will be covered in "getFeatureVariableValueForType" task where variable usage will be tested.
For now in order to increase code coverage I added basic test cases for these in OPTLYProjectConfigTest class under "Test [OPTLYVariation getVariableUsageForVariableId]:" mark. Now it is covered 100%